src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -1095,6 +1097,13 @@ private static unsafe void UInt32ToNumber(uint value, ref NumberBuffer number)
         private static unsafe string UInt32ToDecStr(uint value, int digits)
         {
             int bufferLength = Math.Max(digits, FormattingHelpers.CountDigits(value));
```

Comment:
Why do we have to call CountDigits, when we have a small 'value' 0-10 ?
Can't we first test the value and see it is range. [0-10]

Reply:
CountDigits is generally quite fast already and in the 0-10 case, it would be just 2 comparison instead of one (without adding the overhead of the 0-10 check for all other values):
https://github.com/dotnet/coreclr/blob/d2e2299a9ea141eaffc0f2c1bee0479e052dd7c7/src/System.Private.CoreLib/shared/System/Buffers/Text/FormattingHelpers.CountDigits.cs#L69